website/project/views/comment.py

```diff
@@ -15,104 +15,214 @@
 from website.notifications.emails import notify
 from website.filters import gravatar
 from website.models import Guid, Comment
+from website.addons.base import GuidFile
 from website.project.decorators import must_be_contributor_or_public
 from datetime import datetime
 from website.project.model import has_anonymous_link
+from website.project.views.node import _view_project, n_unread_comments
 
 
-def resolve_target(node, guid):
+@must_be_contributor_or_public
+def view_comments_project(**kwargs):
+    """
```

Comment:
Don't forget to fill in the docstrings. There seems to be several empty blocks like this one.